x-pack/packages/ml/aiops_log_rate_analysis/state/log_rate_analysis_field_candidates_slice.ts

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { PayloadAction } from '@reduxjs/toolkit';
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+
+import { i18n } from '@kbn/i18n';
+import type { HttpSetup, HttpFetchOptions } from '@kbn/core/public';
+import { AIOPS_API_ENDPOINT } from '@kbn/aiops-common/constants';
+
+import type { AiopsLogRateAnalysisSchema } from '../api/schema';
+import type { FetchFieldCandidatesResponse } from '../queries/fetch_field_candidates';
+
+const ecsIdentifiedMessage = i18n.translate(
+  'xpack.aiops.logRateAnalysis.fieldCandidates.ecsIdentifiedMessage',
+  {
+    defaultMessage: 'The source documents were identified as ECS compliant.',
+  }
+);
+
+const fieldsDropdownHintMessage = i18n.translate(
+  'xpack.aiops.logRateAnalysis.fieldCandidates.fieldsDropdownHintMessage',
+  {
+    defaultMessage: 'Use the "Fields" dropdown to adjust the selection.',
+  }
+);
+
+const getFieldSelectionMessage = (
+  isECS: boolean,
+  allItemsCount: number,
+  selectedItemsCount: number
+): string | undefined => {
+  if (allItemsCount <= selectedItemsCount || selectedItemsCount < 2) return;
+
+  const ecsMessage = isECS ? `${ecsIdentifiedMessage} ` : '';
+
+  const fieldsSelectedMessage = i18n.translate(
+    'xpack.aiops.logRateAnalysis.fieldCandidates.fieldsSelectedMessage',
+    {
+      defaultMessage:
+        '{selectedItemsCount} out of {allItemsCount} fields were preselected for the analysis.',
```

Comment:
Is it impossible that there would ever be just 1 field preselected for analysis? If not, the singular / plural check should be added for `fields` in this message.

Reply:
For non-ECS indices this message will only be shown if the index has 100+ fields (saying `100 out of 123 fields were preselected` then). In the weird edge case when an ECS index would have just one field, then this field will be `ecs.version` which is part of the fields we preselect. So we'll select `1 out of 1 field`, but if all fields are selected, then we'll hide this message. tl;dr: The singular case cannot happen with this message.